annotation/chart/templates/deployment.yaml

```diff
@@ -4,40 +4,61 @@ metadata:
   annotations:
     deployment.kubernetes.io/revision: "1"
   labels:
-    app: annotation
-  name: annotation
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}"
+    app: "{{ template "svc.name" . }}"
+    {{- range $key, $value := .Values.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  name: {{ template "svc.name" . }}
   namespace: {{ .Release.Namespace }}
 spec:
   progressDeadlineSeconds: 600
-  replicas: 1
+  replicas: {{ default .Values.global.replicaCount .Values.replicaCount }}
   revisionHistoryLimit: 10
   selector:
     matchLabels:
-      app: annotation
+      app: {{ template "svc.name" . }}
   strategy:
-    rollingUpdate:
-      maxSurge: 25%
-      maxUnavailable: 25%
-    type: RollingUpdate
+{{ .Values.updateStrategy | default .Values.global.updateStrategy | toYaml | indent 4 }}
   template:
     metadata:
-      annotations:
-        sidecar.istio.io/inject: "false"
-      creationTimestamp: null
       labels:
-        app: annotation
+        release: {{ .Release.Name | quote }}
+        chart: "{{ .Chart.Name }}"
+        app: "{{ template "svc.name" . }}"
+        {{- if .Values.labels }}
+        {{- range $key, $value := .Values.labels }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+        {{- else }}
+        {{- range $key, $value := .Values.global.labels }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+        {{- end }}
+      annotations:
+        {{- if .Values.podAnnotation }}
+        {{- range $key, $value := .Values.podAnnotations }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+        {{- else}}
+        {{- range $key, $value := .Values.global.podAnnotations }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+        {{- end }}
     spec:
+      affinity:
+{{ .Values.affinity | default .Values.global.affinity | toYaml | indent 8 }}
+      nodeSelector:
+{{ .Values.nodeSelector | default .Values.global.nodeSelector | toYaml | indent 8 }}
+      tolerations:
+{{ .Values.tolerations | default .Values.global.tolerations | toYaml | indent 8 }}
       containers:
-      - image: {{ .Values.image.registry }}/annotation:{{ .Values.image.tag }}
-        imagePullPolicy: IfNotPresent
-        name: annotation
+      - image: "{{ .Values.imageName }}:{{ default .Values.global.imageTag .Values.imageTag }}"
+        imagePullPolicy: "{{ default .Values.global.imagePullPolicy .Values.imagePullPolicy }}"
+        name: {{ template "svc.name" . }}
         resources:
-          requests:
-            memory: "200Mi"
-            cpu: "200m"
-          limits:
-            memory: "1000Mi"
-            cpu: "400m"
+{{ .Values.resources | default .Values.global.resources | toYaml | indent 10 }}
```

Comment:
I'm confused, what's going on here?
If I define resources in global value file, I will get the same resource value for each app. I would prefere to have the ability to manage resources for each app.
If I don't define  resources in global values, I will get resources: null. 
Example:

global values.yaml
```
#  resources: {}
#    requests:
#      memory: "1000Mi"
#      cpu: "000m"
#    limits:
#      memory: "0000Mi"
#      cpu: "000m"
```

values.yaml
```
resources:
    requests:
        memory: "200Mi"
        cpu: "200m"
    limits:
        memory: "1000Mi"
        cpu: "400m"
 ```
        
 Output:
 ```
   - image: "818863528939.dkr.ecr.eu-central-1.amazonaws.com/badgerdoc/users:0.1.2-0d3e100"
        imagePullPolicy: "Always"
        name: users
        resources:
          null
          ```